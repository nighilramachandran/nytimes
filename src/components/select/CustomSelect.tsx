import {
  FormControl,
  InputLabel,
  MenuItem,
  Select,
  SelectChangeEvent,
} from "@mui/material";
import React from "react";

// interfaces
export interface customSlectProps {
  options: string[];
  defaultOption: string;
  optionChange: (val: string) => void;
}

const CustomSelect: React.FC<customSlectProps> = ({
  options,
  defaultOption,
  optionChange,
}) => {
  // functions

  const handleSelectChange = (event: SelectChangeEvent) => {
    optionChange && optionChange(event.target.value as string);
  };
  return (
    <FormControl sx={{ width: "200px" }} data-testid="form-control-select">
      <InputLabel
        sx={{ color: "text.primary" }}
        data-testid="form-control-label"
      >
        Select Time Period
      </InputLabel>
      <Select
        value={defaultOption}
        className="outlined-gradient"
        onChange={(e) => handleSelectChange(e)}
        data-testid="form-control-select"
      >
        {options?.map((option, index) => {
          return (
            <MenuItem
              key={option}
              value={option}
              data-testid={`"form-control-select-menu-item-${index}"`}
            >
              {option}
            </MenuItem>
          );
        })}
      </Select>
    </FormControl>
  );
};

export default CustomSelect;
