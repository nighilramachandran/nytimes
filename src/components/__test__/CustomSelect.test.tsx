import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import CustomSelect, { customSlectProps } from "../select/CustomSelect";
import "@testing-library/jest-dom/extend-expect";

// Mock props for testing
const mockProps: customSlectProps = {
  options: ["Option 1", "Option 2", "Option 3"],
  defaultOption: "Option 1",
  optionChange: jest.fn(), // Mock function
};

describe("CustomSelect Component", () => {
  test("renders all menu items based on options prop", () => {
    render(<CustomSelect {...mockProps} />);

    // Check if all menu items are rendered correctly
    mockProps.options.forEach((option, index) => {
      const menuItem = screen.queryByTestId(
        `form-control-select-menu-item-${index}`
      );
      if (menuItem) {
        // eslint-disable-next-line jest/no-conditional-expect
        expect(menuItem.textContent).contain(option);
        // eslint-disable-next-line jest/no-conditional-expect
        expect(menuItem.getAttribute("value")).to(option);
      }
    });
  });
});
