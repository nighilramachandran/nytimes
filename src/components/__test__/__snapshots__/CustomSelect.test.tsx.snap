// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect Component matches snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1orw215-MuiFormControl-root"
    data-testid="form-control-select"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-52454s-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      data-testid="form-control-label"
    >
      Select Time Period
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl outlined-gradient css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="form-control-select"
    >
      <div
        aria-controls=":r4:"
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="combobox"
        tabindex="0"
      >
        Option 1
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="Option 1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
